util-core/src/test/java/com/twitter/util/FunctionCompilationTest.java

```diff
@@ -34,11 +36,66 @@ public Integer applyE() throws Exception {
         throw new Exception("Expected");
       }
     };
-    try {
-      fun.apply();
-      assert false : "Should have thrown";
-    } catch (Exception e) {
-      // pass: expected
-    }
+    fun.apply();
+  }
+
+  @Test
+  public void testMakeFunctionFromLambda() {
```

Comment:
I think that's how they already work - they are compiled but not run. I commented out the Java 8 part since the code needs to be able to be compiled with JDK 7. (But if you un-outcomment it, then it should compile.) I was asking on [stackoverflow](http://stackoverflow.com/questions/27881595/how-to-set-up-build-to-accept-src-test-java8-only-when-java-8-is-used) if you can configure sbt to only compile java 8 files if running with jdk 8, so you can set up a Travis CI cross-build - until then, we cannot have actual lambda expressions here (unfortunately).